pkg/utils/errors/errors.go

```diff
@@ -0,0 +1,112 @@
+package errors
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+)
+
+// repoRoot is used to limit frame processing to only those created by this project.  See frameUnstack() for why it is hardcoded
+const repoRoot = "containerized-data-importer"
+
+// prettyError wraps a passed in message and the internally generated stack trace.  `message` may be
+// an string or an error interface.
+type prettyError struct {
+	message    string
+	traceStack *trace
+	frameStack []runtime.Frame
+}
+
+// Error satisfies the built-in error interface
+func (e prettyError) Error() string {
+	return fmt.Sprintf("ERROR: %v\n%s", e.message, e.traceStack)
+}
+
+// String satisfies the built-in Stringer interface
+func (e prettyError) String() string {
+	return e.Error()
+}
+
+// Err is intended to behave as the built-in errors.New() function.  Use this function to generate simple prettyErrors.
+// Some pre-formatting is performed by trimming trailing new lines.
+func Err(msg string) *prettyError {
+	return newError(strings.TrimRight(msg, "\n"), 3)
+}
+
+// Errf is intended to behave as the built-in fmt.Errorf() function. Use this function to generate formatted prettyErrors.
+// Some pre-formatting is performed by trimming trailing new lines.
+func Errf(pattern string, args ...interface{}) *prettyError {
+	return newError(strings.TrimRight(fmt.Sprintf(pattern, args...), "\n"), 3)
```

Comment:
It could be a constant, but it would still have to be passed in to `newErr`.  If new `newErr` were exported as replied above, however, this would mean that we couldn't wrap `NewErr` with a formatting version (`Errf`) since `NewErr` would have a skip of 2, while `Errf` would have a skip of 3.

Reply:
agree